compiler/rustc_mir_transform/src/early_otherwise_branch.rs

```diff
@@ -95,7 +95,7 @@ pub struct EarlyOtherwiseBranch;
 
 impl<'tcx> MirPass<'tcx> for EarlyOtherwiseBranch {
     fn is_enabled(&self, sess: &rustc_session::Session) -> bool {
-        sess.mir_opt_level() >= 2
+        sess.mir_opt_level() >= 3 && sess.opts.debugging_opts.unsound_mir_opts
```

Comment:
Is changing the opt level from 2 -> 3 necessary? The check for `unsound_mir_opts` would be sufficient, I think.

Reply:
I don't think so, I'm just mostly mirroring what I've seen elsewhere